vm/src/hint_processor/builtin_hint_processor/bootloader_hints.rs

```diff
@@ -0,0 +1,191 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+use crate::hint_processor::builtin_hint_processor::hint_utils::insert_value_from_var_name;
+use crate::hint_processor::hint_processor_definition::HintReference;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::exec_scope::ExecutionScopes;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::errors::vm_errors::VirtualMachineError;
+use crate::vm::runners::builtin_runner::OutputBuiltinRunner;
+use crate::vm::vm_core::VirtualMachine;
+
+/// Names for internal variables stored in the execution scope.
+mod vars {
+    /// Deserialized bootloader input.
+    pub(crate) const BOOTLOADER_INPUT: &str = "bootloader_input";
+
+    /// Saved state of the output builtin.
+    pub(crate) const OUTPUT_BUILTIN_STATE: &str = "output_builtin_state";
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(transparent)]
+pub struct ProgramHash(pub u64);
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct BootloaderConfig {
+    pub simple_bootloader_program_hash: ProgramHash,
+    pub supported_cairo_verifier_program_hashes: Vec<ProgramHash>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PackedOutput {
+    // TODO: missing definitions of PlainPackedOutput, CompositePackedOutput
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct BootloaderInput {
+    pub bootloader_config: BootloaderConfig,
+    pub packed_outputs: Vec<PackedOutput>,
+}
+
+fn replace_output_builtin(
+    vm: &mut VirtualMachine,
+    mut new_builtin: OutputBuiltinRunner,
+) -> Result<OutputBuiltinRunner, VirtualMachineError> {
+    let old_builtin = vm.get_output_builtin()?;
+    std::mem::swap(old_builtin, &mut new_builtin);
+    Ok(new_builtin)
+}
+
+/// Implements
+/// %{
+///     from starkware.cairo.bootloaders.bootloader.objects import BootloaderInput
+///     bootloader_input = BootloaderInput.Schema().load(program_input)
+///
+///     ids.simple_bootloader_output_start = segments.add()
+///
+///     # Change output builtin state to a different segment in preparation for calling the
+///     # simple bootloader.
+///     output_builtin_state = output_builtin.get_state()
+///     output_builtin.new_state(base=ids.simple_bootloader_output_start)
+/// %}
+pub fn prepare_simple_bootloader_output_segment(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    // Python: bootloader_input = BootloaderInput.Schema().load(program_input)
```

Comment:
I like these annotations about the "old" hint code :+1:

Reply:
Yeah I think that it makes things clearer when implementing longer hints.